### PR TITLE
auth1.py

### PR DESCRIPTION
**Pull request by pythexcel**

### CODE REVIEW

File: app/api/auth.py [hunk 1]
```diff
@@ -17,57 +17,62 @@
...
-            return jsonify(msg="invalid password"), 500
+            return jsonify(msg="invalid password"),500  
     else:
         return jsonify(msg="invalid login"), 500
 
 
+@bp.route('/update_profile/<string:id>', methods=['PUT'])
+@jwt_required
+@token.admin_required
+def update_profile(id):
```
Comment (pythexcel) [inline app/api/auth.py]: what is logic of this? when do u think frontend will call this?

File: app/api/auth.py [hunk 2]
```diff
@@ -17,57 +17,62 @@
...
-            access_token = create_access_token(
-                identity=user, expires_delta=expires)
+    
+   
+    payload = {'username': username, "password": password, "action": "login", "token": None}
+    response = requests.post(url=URL, json=payload)
+    token = response.json()
+    if response is not None:
+        URL2 = 'https://hr.excellencetechnologies.in/attendance/sal_info/api.php'
+        payload2 = {"action": "get_user_profile_detail", "token": token['data']['token']}
```
Comment (pythexcel) [inline app/api/auth.py]: payload2, response2. are not proper variable names. 
payload_user_details would have been better

Comment (pythexcel) [reply app/api/auth.py]: make proper variable names

File: app/api/auth.py [hunk 3]
```diff
@@ -17,57 +17,62 @@
...
-        if pbkdf2_sha256.verify(password, user["password"]):
-            expires = datetime.timedelta(days=1)
-            access_token = create_access_token(
-                identity=user, expires_delta=expires)
+    
+   
+    payload = {'username': username, "password": password, "action": "login", "token": None}
+    response = requests.post(url=URL, json=payload)
+    token = response.json()
+    if response is not None:
```
Comment (pythexcel) [inline app/api/auth.py]: where is else condition? we cannot write a production code without else

File: app/api/auth.py [hunk 4]
```diff
@@ -17,57 +17,62 @@
...
+        URL2 = 'https://hr.excellencetechnologies.in/attendance/sal_info/api.php'
+        payload2 = {"action": "get_user_profile_detail", "token": token['data']['token']}
+        response2 = requests.post(url=URL2, json=payload2)
+        username = request.json.get("username", None)
+        result = response2.json()
+        user = mongo.db.users.count({
+            "username": username})
+        if user > 0:
+                return jsonify({"msg": "Username already taken"}), 500
```
Comment (pythexcel) [inline app/api/auth.py]: this is wrong. logic. think!

this is not a create account router. this is login route.

after first time, username will be taken for sure. we will not allow user to login??

File: app/api/auth.py [hunk 5]
```diff
@@ -17,57 +17,62 @@
...
+
+        user = mongo.db.users.insert_one({
+            "profile": result,
+           "username": username
+        }).inserted_id
+        
+        if username is not None:
+            access_token = create_access_token(identity=username)
```
Comment (pythexcel) [inline app/api/auth.py]: this login route.
where is logic for wrong user/pass
invalid user etc?

File: app/api/auth.py [hunk 6]
```diff
@@ -17,56 +17,90 @@
...
+    token = response_user_token.json()
+    if token['data'] == {'message': 'Invalid Login'}:
+        return jsonify(msg='invalid login')
     else:
-        return jsonify(msg="invalid login"), 500
-
+        URL_details = 'https://hr.excellencetechnologies.in/attendance/sal_info/api.php'
+        payload_user_details = {"action": "get_user_profile_detail", "token": token['data']['token']}
+        response_user_details = requests.post(url=URL_details, json=payload_user_details)
+        username = request.json.get("username", None)
```
Comment (xmagestore) [inline app/api/auth.py]: we already have this in variable why again

File: app/api/settings.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+
+@bp.route('/settings', methods=['POST'])
+def settings():
+ if not request.json:
+     abort(500)
+ integrate_with_hr = request.json.get('integrate_with_hr', False)
+ hr = mongo.db.hr.count({
+     "integrate_with_hr": integrate_with_hr})
+
```
Comment (reactexcel) [inline app/api/settings.py]: no. wrong. just do update with insert=true below will happen in one query only

File: app/api/auth.py [hunk 8]
```diff
@@ -19,54 +19,135 @@
...
-
-    id = mongo.db.users.insert_one({
-        "name": name,
-        "password": pbkdf2_sha256.hash(password),
-        "username": username
-    }).inserted_id
-    return jsonify(str(id))
+   if not request.json:
+       abort(500)
+   name = request.json.get("name", None)
```
Comment (reactexcel) [inline app/api/auth.py]: there is no condition here. e.g registration route should not work if the setting is on? 
same for login

File: app/api/auth.py [hunk 9]
```diff
@@ -19,54 +19,142 @@
...
-    user = mongo.db.users.count({
-        "username": username
-    })
-    if user > 0:
-        return jsonify({"msg": "Username already taken"}), 500
+   hr = mongo.db.hr.find_one({
+       "integrate_with_hr": True
+   })
+   if hr is not None and "integrate_with_hr" in hr:
+
```
Comment (reactexcel) [inline app/api/auth.py]: why this space. status code should be 500 not 200

**Pull request merged**